src/platforms/shared.rs

```diff
@@ -82,3 +81,21 @@ pub fn shared_build_opts_ccache(use_ccache: bool) -> Vec<String> {
         vec![]
     }
 }
+
+/// Ignore duplicate libraries warnings for MacOS with XCode>=15.
+pub fn macos_build_opts_ignore_dupicate_libs_warnings() -> Vec<String> {
+    let xcode_version =
+        crate::utils::get_xcode_version().unwrap_or(crate::utils::XCODE_MIN_VERSION);
+    if xcode_version >= 15 {
```

Comment:
Let's move this magic number to constant as well.

Reply:
Done.